### ISSUE TITLE
Wrong keras acc?

### ISSUE THREAD

**Issue by otblearn** [fchollet/keras#5210]
Hey,

I have a small model to solve a multi classification problem. While training and testing the model accuracy is around ~0.96, but when I check the prediction of the model, it seems to be very wrong... 

    top_words = 5000
    max_review_length = 600
    embedding_vector_length = 64

    model = Sequential()
    model.add(Embedding(top_words, embedding_vector_length, input_length=max_review_length, dropout=0.2))
    model.add(Dropout(0.2))
    model.add(LSTM(100))
    model.add(Dropout(0.2))
    model.add(Dense(30, activation='sigmoid'))
    model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
  
    model.fit(X_train, y_train, nb_epoch=1, batch_size=64)

    scores = model.evaluate(X_test, y_test)
    print("%s: %.2f%%" % (model.metrics_names[1], scores[1] * 100))

    preds = model.predict(X_test)
    print preds[:5]
    print y_test[:5]

**Comment by otblearn** [comment]
Thanks man!
